src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Testing/RunTestsHandler.cs

```diff
@@ -160,4 +165,34 @@ private static TraceLevel GetTraceLevel(ServerConfiguration serverConfiguration)
             _ => throw new InvalidOperationException($"Unexpected log level {serverConfiguration.MinimumLogLevel}"),
         };
     }
+
+    private async Task<string?> GetRunSettingsAsync(string? runSettingsPath, BufferedProgress<RunTestsPartialResult> progress, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrEmpty(runSettingsPath))
+        {
+            return null;
+        }
+
+        if (!File.Exists(runSettingsPath))
+        {
+            var message = string.Format(LanguageServerResources.Runsettings_file_does_not_exist_at_0, runSettingsPath);
+            progress.Report(new(LanguageServerResources.Discovering_tests, message, Progress: null));
+            return null;
+        }
+
+        try
+        {
+            var contents = await File.ReadAllTextAsync(runSettingsPath, cancellationToken);
```

Comment:
Should this line be wrapped in `IOUtilities.PerformIOAsync`?

Reply:
Hmm not sure.  I think I would only catch and log the exception regardless of what kind it is?  Not sure if I would do anything different if it wasn't a normal IO exception?